OpenSim/Actuators/MuscleFixedWidthPennationModel.h

```diff
@@ -342,19 +325,27 @@ OpenSim_DECLARE_CONCRETE_OBJECT(MuscleFixedWidthPennationModel, Object);
                              double muscleVelocity,
                              double tendonVelocity) const;
 
-    void ensureModelUpToDate();
+protected:
+    // Component interface.
+    void extendFinalizeFromProperties() override;
 
 private:
     void setNull();
     void constructProperties();
-    void buildModel();
-    void ensurePropertiesSet() const;
 
     double m_parallelogramHeight;
     double m_maximumSinPennation;
     double m_minimumFiberLength;
     double m_minimumFiberLengthAlongTendon;
 
+    // Enforces a lower bound on the fiber length to avoid a numerical
+    // singularity as the fiber length approaches zero.
+    double clampFiberLength(double fiberLength) const;
+
+    // These classes are friends because they call clampFiberLength().
+    friend class Thelen2003Muscle;
+    friend class Millard2012EquilibriumMuscle;
+    friend class Millard2012AccelerationMuscle;
```

Comment:
>That seems to introduce inconsistencies in fiberLength.

I agree. Note that `clampFiberLength()` returns `max(m_minimumFiberLength, fiberLength)`. Does this mean that fiber lengths smaller than `m_minimumFiberLength` could be stored in the states file, despite not being used by the muscle model?